Attach tests fail on Mac OS

```
Traceback (most recent call last):
  File "/Users/vsts/agent/2.134.2/work/1/s/tests/highlevel/test_lifecycle.py", line 107, in test_attach
    self.attach(expected_os_id=OS_ID, attach_args={})
  File "/Users/vsts/agent/2.134.2/work/1/s/tests/highlevel/test_lifecycle.py", line 94, in attach
    startMethod='attach',
  File "/Users/vsts/agent/2.134.2/work/1/s/tests/highlevel/__init__.py", line 901, in assert_vsc_received
    assert_messages_equal(received, expected)
  File "/Users/vsts/agent/2.134.2/work/1/s/tests/helpers/message.py", line 62, in assert_messages_equal
    raise AssertionError('\n'.join(msg))
AssertionError: 
Received:
Event(seq=0, event='output', body={'category': 'telemetry', 'output': 'ptvsd', 'data': {'version': '4.1.1a9+23.g86c9685'}})
Response(seq=1, request_seq=0, command='initialize', success=True, message='', body={'supportsConditionalBreakpoints': True, 'supportsConfigurationDoneRequest': True, 'supportsDebuggerProperties': True, 'supportsEvaluateForHovers': True, 'supportsExceptionInfoRequest': True, 'supportsExceptionOptions': True, 'supportsHitConditionalBreakpoints': True, 'supportsLogPoints': True, 'supportsModulesRequest': True, 'supportsSetExpression': True, 'supportsSetVariable': True, 'supportsValueFormattingOptions': True, 'supportTerminateDebuggee': True, 'exceptionBreakpointFilters': [{'filter': 'raised', 'label': 'Raised Exceptions', 'default': False}, {'filter': 'uncaught', 'label': 'Uncaught Exceptions', 'default': True}]})
Event(seq=2, event='initialized', body={})
Response(seq=3, request_seq=1, command='attach', success=True, message='', body={})
Response(seq=4, request_seq=2, command='configurationDone', success=True, message='', body={})

Expected:
Event(seq=0, event='output', body={'category': 'telemetry', 'output': 'ptvsd', 'data': {'version': '4.1.1a9+23.g86c9685'}})
Response(seq=1, request_seq=0, command='initialize', success=True, message='', body={'supportsConditionalBreakpoints': True, 'supportsConfigurationDoneRequest': True, 'supportsDebuggerProperties': True, 'supportsEvaluateForHovers': True, 'supportsExceptionInfoRequest': True, 'supportsExceptionOptions': True, 'supportsHitConditionalBreakpoints': True, 'supportsLogPoints': True, 'supportsModulesRequest': True, 'supportsSetExpression': True, 'supportsSetVariable': True, 'supportsValueFormattingOptions': True, 'supportTerminateDebuggee': True, 'exceptionBreakpointFilters': [{'filter': 'raised', 'label': 'Raised Exceptions', 'default': False}, {'filter': 'uncaught', 'label': 'Uncaught Exceptions', 'default': True}]})
Event(seq=2, event='initialized', body={})
Response(seq=3, request_seq=1, command='attach', success=True, message='', body={})
Response(seq=4, request_seq=2, command='configurationDone', success=True, message='', body={})
Event(seq=5, event='process', body={'name': '/Users/vsts/agent/2.134.2/work/1/s/tests/__main__.py', 'systemProcessId': 543, 'isLocalProcess': True, 'startMethod': 'attach'})

Diff by line
  0:  Event(seq=0, event='output', body={'category': 'telemetry', 'output': 'ptvsd', 'data': {'version': '4.1.1a9+23.g86c9685'}})
  1:  Response(seq=1, request_seq=0, command='initialize', success=True, message='', body={'supportsConditionalBreakpoints': True, 'supportsConfigurationDoneRequest': True, 'supportsDebuggerProperties': True, 'supportsEvaluateForHovers': True, 'supportsExceptionInfoRequest': True, 'supportsExceptionOptions': True, 'supportsHitConditionalBreakpoints': True, 'supportsLogPoints': True, 'supportsModulesRequest': True, 'supportsSetExpression': True, 'supportsSetVariable': True, 'supportsValueFormattingOptions': True, 'supportTerminateDebuggee': True, 'exceptionBreakpointFilters': [{'filter': 'raised', 'label': 'Raised Exceptions', 'default': False}, {'filter': 'uncaught', 'label': 'Uncaught Exceptions', 'default': True}]})
  2:  Event(seq=2, event='initialized', body={})
  3:  Response(seq=3, request_seq=1, command='attach', success=True, message='', body={})
  4:  Response(seq=4, reque
```

--- a/tests/helpers/debugadapter.py
+++ b/tests/helpers/debugadapter.py
@@ -79,10 +79,12 @@ def wait_for_socket_server(addr, timeout=3.0, **kwargs):
     start_time = time.time()
     while True:
         try:
-            sock = socket.create_connection(addr)
+            print('inside wait')
+            print(addr.host, addr.port)
+            sock = socket.create_connection((addr.host, addr.port))
             sock.close()
             return
-        except ConnectionRefusedError:
+        except Exception:
             pass
         time.sleep(0.1)
         if time.time() - start_time > timeout:

--- a/tests/helpers/debugadapter.py
+++ b/tests/helpers/debugadapter.py
@@ -79,8 +79,6 @@ def wait_for_socket_server(addr, timeout=3.0, **kwargs):
     start_time = time.time()
     while True:
         try:
-            print('inside wait')
-            print(addr.host, addr.port)
             sock = socket.create_connection((addr.host, addr.port))
             sock.close()
             return

--- a/tests/helpers/debugadapter.py
+++ b/tests/helpers/debugadapter.py
@@ -75,7 +75,7 @@ def _copy_env(verbose=False, env=None):
     return variables
 
 
-def _wait_for_socket_server(addr, timeout=3.0, **kwargs):
+def wait_for_socket_server(addr, timeout=3.0, **kwargs):
     start_time = time.time()
     while True:
         try:
@@ -151,7 +151,9 @@ class DebugAdapter(Closeable):
     @classmethod
     def start_for_attach(cls, addr, *args, **kwargs):
         addr = Address.as_server(*addr)
-        return cls._start_as(addr, *args, server=True, **kwargs)
+        adapter = cls._start_as(addr, *args, server=True, **kwargs)
+        wait_for_socket_server(addr)
+        return adapter
 
     @classmethod
     def _start_as(cls, addr, name, kind='script', extra=None, server=False,
@@ -182,7 +184,7 @@ class DebugAdapter(Closeable):
             addr=addr,
             **kwargs
         )
-        _wait_for_socket_server(addr, **kwargs)
+        wait_for_socket_server(addr, **kwargs)
         return adapter
 
     @classmethod

--- a/tests/helpers/debugclient.py
+++ b/tests/helpers/debugclient.py
@@ -5,7 +5,7 @@ import warnings
 from ptvsd.socket import Address
 from ptvsd._util import new_hidden_thread
 from . import Closeable
-from .debugadapter import DebugAdapter
+from .debugadapter import DebugAdapter, wait_for_socket_server
 from .debugsession import DebugSession
 
 
@@ -116,6 +116,7 @@ class _LifecycleClient(Closeable):
         if wait_for_connect:
             wait_for_connect()
         else:
+            wait_for_socket_server(addr)
             self._attach(addr, **kwargs)
 
     def _attach(self, addr, **kwargs):

--- a/tests/highlevel/test_lifecycle.py
+++ b/tests/highlevel/test_lifecycle.py
@@ -64,7 +64,8 @@ class LifecycleTests(HighlevelTest, unittest.TestCase):
                 req_attach = self.send_request('attach', attach_args)
 
             # configuration
-            req_config = self.send_request('configurationDone')
+            with self._fix.wait_for_events(['process']):
+                req_config = self.send_request('configurationDone')
 
             # Normal ops would go here.
 

